test/solver_tests/greedy_variants_test.py

```diff
@@ -160,13 +148,10 @@ def test_maxcut_base_case_weights_trivial(self):
         mcgsolver = MaxCutGreedySolver(
             character_matrix=cm, missing_char=-1, priors=priors
         )
```

Comment:
I believe it's this test that fails sporadically (i.e. there's some randomization with the weighting). There are two solutions I can see:

1) set a seed for this test so you know you'll get the same behavior each time
2) come up with a more deterministic test case.